transport/tlsfrag/writer.go

```diff
@@ -160,7 +166,7 @@ func (w *clientHelloFragWriter) splitHelloBufToRecord() {
 	// splitted: | <= (5) => | <= head => | <= (5) => | <= tail => |
 	//           |  header1  |  payload1  |  header2  |  payload2  |
 	splitted := original[:len(original)+recordHeaderLen]
-	hdr1 := tlsRecordHeaderFromRawBytes(splitted[:recordHeaderLen])
+	hdr1, _ := newTLSHandshakeRecordHeader(splitted[:recordHeaderLen])
```

Comment:
Do you need to check for errors?

Reply:
The errors should be checked in the `helloBuf` itself: https://github.com/Jigsaw-Code/outline-sdk/pull/207/files/5aff31283fb85ade5b32a63d59b29bdf33ec8f97#diff-30f1fffb3a03fa3317f98666620087beb984e6ffc72af10b7634150365ef4e83R96

But I think we can refactor it later to directly returning the `TLSHandshakeRecordHeader` object from the `helloBuf`, similar to what I've done in the `recordLenWriter`.